wandb/sdk/internal/file_stream.py

```diff
@@ -249,43 +306,64 @@ class FileStreamApi:
     TODO: Differentiate between binary/text encoding.
     """
 
-    Finish = collections.namedtuple("Finish", ("exitcode"))
-    Preempting = collections.namedtuple("Preempting", ())
-    PushSuccess = collections.namedtuple("PushSuccess", ("artifact_id", "save_name"))
+    class Finish(NamedTuple):
+        exitcode: int
+
+    class Preempting(NamedTuple):
+        pass
+
+    class PushSuccess(NamedTuple):
+        artifact_id: str
+        save_name: str
 
     HTTP_TIMEOUT = env.get_http_timeout(10)
     MAX_ITEMS_PER_PUSH = 10000
 
-    def __init__(self, api, run_id, start_time, settings=None):
-        if settings is None:
-            settings = dict()
+    def __init__(
+        self,
+        api: "internal_api.Api",
+        run_id: str,
+        start_time: float,
+        settings: Optional[
+            Union[
+                "wandb.sdk.wandb_settings.Settings",
+                "settings_static.SettingsStatic",
+                dict,
+            ]
+        ] = None,
+    ) -> None:
+        settings = settings or dict()
         # NOTE: exc_info is set in thread_except_body context and readable by calling threads
-        self._exc_info = None
+        self._exc_info: Optional[
+            Union[
+                Tuple[Type[BaseException], BaseException, TracebackType],
+                Tuple[None, None, None],
+            ]
+        ] = None
         self._settings = settings
         self._api = api
         self._run_id = run_id
         self._start_time = start_time
-        self._client = requests.Session()
-        self._client.auth = ("api", api.api_key)
-        self._client.timeout = self.HTTP_TIMEOUT
```

Comment:
@raubitsj `requests.Session` does not have a timeout argument, so no timeout has ever been in place here - this has been an ongoing discussion for like 5+ years in the community. I used one of the standard workarounds.